docs/guides/incremental_time.md

```diff
@@ -0,0 +1,156 @@
+- 24 for each hour of yesterday
+- 12 for each hour today from 12am to 12pm
+
+All intervals are complete because it is 12pm (so the 11am interval has ended). This is illustrated in the bottom panel of Figure 2.
+
+When we first execute and backfill the bottom model as part of a `sqlmesh plan` today at 12pm, SQLMesh calculates its set of 60 time intervals and records that all 60 of them were backfilled. It retains this information in the SQLMesh state tables for future use.
+
+If we `sqlmesh run` the model *tomorrow* at 12pm, SQLMesh calculates the set of all intervals as:
+
+- 24 for two days ago
+- 24 for yesterday
+- 24 for today
+- 12 for tomorrow 12am to 12pm
+
+This gives a total of 84 intervals.
+
+It compares this set of 84 to the stored set of 60 that we already backfilled to identify the 24 un-processed intervals from 12pm yesterday to 12pm today. It then processes only those 24 intervals during today's run.
+
+## Running `run`
+
+SQLMesh has two different commands for processing data. If any model has been changed, [`sqlmesh plan`](../reference/cli.md#plan) is used to apply the change to data in a specific environment. If no models have changed, [`sqlmesh run`](../reference/cli.md#run) is used to execute the project's models.
+
+Data accumulation rates and freshness requirements may differ across models. If `sqlmesh run` ran every model whenever it was executed, all models would be held to the same freshness requirements as the most stringent model. This is inefficient and wastes computational resources (and money).
+
+Instead, you specify the [`cron` parameter](../concepts/models/overview.md#cron) for each model. `sqlmesh run` uses each model's `cron` to determine whether that model should be executed in a given run.
+
+For example, if your most frequent model's `cron` is hourly, you need to execute the `sqlmesh run` command at least hourly (with a tool like Linux's cron). That model will run once every hour, but another model with a `cron` of daily will only run once per day when 24 hours have elapsed since its previous run.
+
+### Scheduling computations
+
+By default, SQLMesh processes all intervals that have elapsed since a model's previous run in a single job. If a model's source data is large, you may want to break the computations up into smaller jobs - this is done with the model configuration's `batch_size` parameter.
+
+When `batch_size` is specified, the total number of intervals to process is divided into batches of size `batch_size`, and one job is executed for each batch.
+
+## Model time
+
+Incremental by time models require specification of a time column in their configuration. In addition, their model SQL queries should specify a `WHERE` clause that filters the data on a time range.
+
+This example shows an incremental by time model that could be added to the SQLMesh [quickstart project](../quick_start.md):
+
+``` sql linenums="1"
+MODEL (
+    name sqlmesh_example.new_model,
+    kind incremental_by_time_range (
```

Comment:
[Nit] I think the convention is to capitalize kind

Reply:
capitalize all the key names, or just kind?